controllers/webhook/nim_account_webhook.go

```diff
@@ -0,0 +1,116 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	nimv1 "github.com/opendatahub-io/odh-model-controller/api/nim/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// +kubebuilder:webhook:admissionReviewVersions=v1,path=/validate-nim-opendatahub-io-v1-account,mutating=false,failurePolicy=fail,groups=nim.opendatahub.io,resources=accounts,verbs=create;update,versions=v1,name=validating.nim.account.odh-model-controller.opendatahub.io,sideEffects=None
+
+type nimAccountValidator struct {
+	client client.Client
+}
+
+func NewNimAccountValidator(client client.Client) admission.CustomValidator {
+	return &nimAccountValidator{client: client}
+}
+
+func (v *nimAccountValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (warnings admission.Warnings, err error) {
+	account := obj.(*nimv1.Account)
+
+	log := logf.FromContext(ctx).WithName("NIMAccountValidatingWebhook").
+		WithValues("namespace", account.Namespace, "account", account.Name)
+	log.Info("Validating NIM Account creation")
+
+	err = v.verifySingletonInNamespace(ctx, account)
+	if err != nil {
+		log.Error(err, "Rejecting NIM Account creation because checking singleton didn't pass")
+		return nil, err
+	}
+
+	err = v.verifyAPIKeySecret(ctx, nil, account)
+	if err != nil {
+		log.Error(err, "Rejecting NIM Account creation because checking API key Secret didn't pass")
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+func (v *nimAccountValidator) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) (warnings admission.Warnings, err error) {
+	account := newObj.(*nimv1.Account)
+
+	log := logf.FromContext(ctx).WithName("NIMAccountValidatingWebhook").
+		WithValues("namespace", account.Namespace, "account", account.Name)
+	log.Info("Validating NIM Account update")
+
+	err = v.verifyAPIKeySecret(ctx, oldObj.(*nimv1.Account), account)
+	if err != nil {
+		log.Error(err, "Rejecting NIM Account update because checking API key Secret didn't pass")
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+func (v *nimAccountValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (warnings admission.Warnings, err error) {
+	// For deletion, nothing needs to be validated
+	return nil, nil
+}
+
+func (v *nimAccountValidator) verifyAPIKeySecret(ctx context.Context, oldAccount, newAccount *nimv1.Account) error {
```

Comment:
@xieshenzh @etirelli @mpaulgreen 
Why are we checking the API key Secret with the admission webhook?

Verifying only one Account exists in a namespace was the only requirement of the webhook. See [ADR](https://docs.google.com/document/d/1gvWjshALD3Ec6ifwkhqNkla_9dZGlsgyaTkvocuaN8U/edit?tab=t.0):

![image](https://github.com/user-attachments/assets/f728dbad-6f7e-4539-a6cf-6a6f6cd39c45)

This webhook duplicates code that should (and does) exist in the controller. See #289.

Reply:
But the controller also checks if the Secret exists—that's the duplication. The controller will update a status condition on the Account, indicating that the API key secret doesn't exist. If users create it after seeing the status, everything will work.

The API mandates the `spec.apiKeySecret`; shouldn't this be enough for users to understand that they must create a Secret?

Reply:
Checks for the Secret is deleted.